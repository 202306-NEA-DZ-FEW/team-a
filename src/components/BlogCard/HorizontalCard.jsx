import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";
import { IoIosArrowDropright } from "react-icons/io";

import { formatDate, truncateString } from "@/lib/helpers";

function HorizontalCard({ title, description, imageUrl, createdAt, id }) {
  const { t } = useTranslation();
  const date = formatDate(createdAt, "MMM YYYY");
  const truncatedDescription = truncateString(description, 200);
  return (
    <div className='card lg:card-side bg-base-100 shadow-xl h-full lg:h-[26rem]'>
      <figure className=' lg:w-[45%]'>
        <Image
          height={400}
          width={400}
          priority
          src={imageUrl}
          alt='Blog Image'
          className='w-full h-full'
        />
      </figure>
      <div className='card-body lg:w-[35%] p-6 lg:p-10 gap-4'>
        <h2 className='card-title text-2xl lg:text-3xl' title={title}>
          {title}
        </h2>
        <p
          className=' text-md lg:text-xl tracking-wide font-light'
          title={description}
        >
          {truncatedDescription}
        </p>
        <div className='card-actions flex gap-2 justify-between'>
          <span>{date}</span>
          <Link
            href={`/blogs/${id}`}
            className='text-secondary font-bold flex items-center gap-2'
          >
            {t("common:buttons:readMore")}

            <IoIosArrowDropright className='text-xl' />
          </Link>
        </div>
      </div>
    </div>
  );
}
export default HorizontalCard;
