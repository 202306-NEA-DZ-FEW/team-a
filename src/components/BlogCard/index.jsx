import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";

import { formatDate, truncateString } from "@/lib/helpers";

function BlogCard({ id, title, imageUrl, createdAt, description }) {
  const { t } = useTranslation();
  const date = formatDate(createdAt, "MMM YYYY");
  const truncatedDescription = truncateString(description, 60);
  const truncatedTitle = truncateString(title, 20);
  return (
    <div className='card bg-white rounded-xl shadow-lg w-64'>
      <div className='relative'>
        <Image
          className='w-full relative object-cover h-44 rounded-tr-2xl rounded-tl-2xl p-1'
          src={imageUrl}
          alt='Blog Image'
          height={400}
          width={400}
          priority
        />
        <span className='absolute left-1 bottom-1 text-white bg-primary px-2 py-1'>
          {date}
        </span>
      </div>

      <div className='p-3 gap-1 flex flex-col'>
        <h2 className='text-lg font-bold mb-2' title={title}>
          {truncatedTitle}
        </h2>
        <p className='text-sm font-light mb-4' title={description}>
          {truncatedDescription}
        </p>
        <div className='flex justify-end'>
          <Link
            href={`/blogs/${id}`}
            className='text-secondary/[.8] hover:text-secondary font-bold'
          >
            {t("common:buttons:readMore")}
          </Link>
        </div>
      </div>
    </div>
  );
}

export default BlogCard;
