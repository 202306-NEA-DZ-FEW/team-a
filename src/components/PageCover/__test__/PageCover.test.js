import renderer from "react-test-renderer";

import BlogCard from "..";

it("renders correctly", () => {
  const tree = renderer.create(<BlogCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
