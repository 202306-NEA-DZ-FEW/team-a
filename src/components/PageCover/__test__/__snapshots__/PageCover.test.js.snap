// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="hero min-h-screen"
  style={
    Object {
      "backgroundImage": "url(undefined)",
    }
  }
>
  <div
    className="hero-overlay bg-opacity-50"
  />
  <div
    className="hero-content text-center text-neutral-content"
  >
    <div
      className="max-w-md"
    >
      <h1
        className="mb-5 text-5xl font-bold"
      />
      <p
        className="mb-5"
      >
         
      </p>
    </div>
  </div>
</div>
`;
